lib/content-services/src/lib/tree/components/tree.component.spec.ts

```diff
@@ -0,0 +1,259 @@
+/*!
+ * @license
+ * Copyright 2019 Alfresco Software, Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { TreeComponent } from './tree.component';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { CoreTestingModule, PaginationModel, UserPreferencesService } from '@alfresco/adf-core';
+import { MatTreeModule } from '@angular/material/tree';
+import { TreeNode, TreeNodeType } from '../models/tree-node.interface';
+import { singleNode, treeNodesChildrenMockExpanded, treeNodesMock, treeNodesMockExpanded } from '../mock/tree-node.mock';
+import { of } from 'rxjs';
+import { TreeService } from '../services/tree.service';
+import { TreeServiceMock } from '../mock/tree-service.service.mock';
+import { By } from '@angular/platform-browser';
+import { SelectionChange } from '@angular/cdk/collections';
+
+describe('TreeComponent', () => {
+    let fixture: ComponentFixture<TreeComponent<TreeNode>>;
+    let component: TreeComponent<TreeNode>;
+    let userPreferenceService: UserPreferencesService;
+
+    const getDisplayNameValue = (nodeId: string) =>
+        fixture.nativeElement.querySelector(`.mat-tree-node[data-automation-id="node_${nodeId}"] .adf-tree-cell-value`).innerText.trim();
+
+    const getNodePadding = (nodeId: string) => {
+        const element = fixture.nativeElement.querySelector(`.mat-tree-node[data-automation-id="node_${nodeId}"]`);
+        return parseInt(window.getComputedStyle(element).paddingLeft, 10);
+    };
+
+    const getNodeSpinner = (nodeId: string) => fixture.nativeElement.querySelector(`.mat-tree-node[data-automation-id="node_${nodeId}"] .mat-progress-spinner`);
+
+    const getExpandCollapseBtn = (nodeId: string) => fixture.nativeElement.querySelector(`.mat-tree-node[data-automation-id="node_${nodeId}"] .adf-icon`);
```

Comment:
should be probably addressing a button, not the icon inside?

Reply:
Click is now applied to the icon since apart from the icon material spinner is rendered inside when node is loading